UI.Tests/WinCheckerTests.cs

```diff
@@ -0,0 +1,405 @@
+using UI.Enums;
+using UI.Entities;
+using FluentAssertions;
+
+namespace UI.Tests;
+
+public class WinCheckerTests
+{
+    #region checking rows
+
+    [Fact]
+    public void HasWon_WhenThreeMatchingMarksInFirstRow_ShouldReturnTrue()
+    {
+        // arrange
+        Player host = new()
+        {
+          ConnectionId = "player one connection hub id",
+          Name = "Dan",
+          ImageUrl = "https://www.pngall.com/wp-content/uploads/12/Avatar-Profile-Vector.png",
+          Mark = Marks.X,
+          HasTurn = true
+        };
+        string GameId = "g00";
+
+        Marks markToCheck = Marks.X;
+        Marks[] gameBoard = new Marks[] { Marks.X, Marks.X, Marks.X, Marks.X, Marks.X, Marks.O, Marks.X, Marks.X, Marks.X };
+
+        // act
+        Game winChecker = new Game(GameId, host);

```

Comment:
Rename variable to `game` and move to the `// arrange` section. Also in the arrange section, get the internal game board to the state you want by calling `game.Move(x)` as many times as necessary to get the board to the state you want it in.

You can also create helped methods to help with that setup.